src/Symfony/Component/Messenger/Middleware/HandlesRecordedMessagesMiddleware.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Middleware;
+
+use Symfony\Component\Messenger\Exception\MessageRecorderHandlerException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\MessageRecorderInterface;
+
+/**
+ * A middleware that takes all recorded messages and dispatch them to the bus.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+class HandlesRecordedMessagesMiddleware implements MiddlewareInterface
+{
+    private $messageRecorder;
+    private $messageBus;
+
+    public function __construct(MessageBusInterface $messageBus, MessageRecorderInterface $messageRecorder)
+    {
+        $this->messageRecorder = $messageRecorder;
+        $this->messageBus = $messageBus;
+    }
+
+    public function handle($message, callable $next)
+    {
+        // Make sure the recorder is empty before we begin
+        $this->messageRecorder->erase();
```

Comment:
Why would you? 🤔

Reply:
To avoid any side effects. See https://github.com/symfony/symfony/pull/27844#discussion_r200204703